project/Boilerplate.scala

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import scala.collection.mutable.ArrayBuffer
+
+/** This is a simplified version of https://github.com/milessabin/shapeless/blob/80d26bc132139eb887fe1e8d39abbdb8dcb20117/project/Boilerplate.scala
+  *
+  * Generate a range of boilerplate classes, those offering alternatives with 0-22 params
+  * and would be tedious to craft by hand
+  */
+object Boilerplate {
+
```

Comment:
I'm not super keen on adding a code generation step unless it's something we're going to need to update a lot. I have a feeling this is something we'll only need to update sparingly so I'm inclined to instead say we just write this is a scala-cli script, run it once, and commit the generated file. We can include the source code for the scala-cli script in this PR so that if we need to update the file we can. But my current opinion is that we should not include this code gen step unless we find ourselves needing to update this a lot.